rust/src/nfs/nfs2_records.rs

```diff
@@ -96,3 +96,159 @@ pub fn parse_nfs2_attribs(i: &[u8]) -> IResult<&[u8], Nfs2Attributes> {
+            0x00, 0x00, 0x20, 0x00,                         // total count
+        ];
+
+        let (_, handle) = parse_nfs2_handle(buf).expect("Parsing nfs2_handle failed!");
+        assert_eq!(handle.value, &buf[..32]);
+
+        let result = parse_nfs2_request_read(buf).unwrap();
+        match result {
+            (r, request) => {
+                assert_eq!(r.len(), 4);
+                assert_eq!(request.handle, handle);
+                assert_eq!(request.offset, 0);
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_reply_read() {
+
+        let buf: &[u8] = &[
+        // Status: NFS_OK - (0)
+            0x00, 0x00, 0x00, 0x00,
+        // attr_blob
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x81, 0xa4,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x0b,
+            0x00, 0x00, 0x40, 0x00, 0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x02, 0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xb2, 0x5d, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x0b, 0x71, 0xb0, 0x38, 0x47, 0x71, 0xc4,
+            0x00, 0x08, 0xb2, 0x90, 0x38, 0x47, 0x75, 0xea,
+            0x00, 0x09, 0x00, 0xb0,
+        // data_len: (11)
+            0x00, 0x00, 0x00, 0x0b,
+        // data_contents: ("the b file")
+            0x74, 0x68, 0x65, 0x20,
+            0x62, 0x20, 0x66, 0x69, 0x6c, 0x65, 0x0a,
+        // _data_padding: parsed as part of data_contents
+            0x00,
+        ];
+
+        let result = parse_nfs2_reply_read(buf).unwrap();
+        match result {
+            (r, response) => {
+                assert_eq!(r.len(), 0);
+                assert_eq!(response.status, 0);
+                assert_eq!(response.attr_follows, 1);
+                assert_eq!(response.attr_blob.len(), 68);
+                assert_eq!(response.count, response.data_len);
+                assert_eq!(response.eof, false);
+                assert_eq!(response.data_len, 11);
+                assert_eq!(response.data, "the b file\n\0".as_bytes());
+                assert_eq!(response.data, &buf[76..]);
+            }
+        }
+    }
+
+    #[test]
+    fn test_nfs2_attributes() {
+
+        let buf: &[u8] = &[
+        // Type: Regular File (1)
+            0x00, 0x00, 0x00, 0x01,
+        // Attributes
+        // _blob1
+            0x00, 0x00, 0x81, 0xa4,
+            0x00, 0x00, 0x00, 0x01,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x01,
+        // Size: (0)
+            0x00, 0x00, 0x00, 0x00,
+        // _blob2
+            0x00, 0x00, 0x40, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x00, 0x00, 0x00,
+            0x00, 0x10, 0x10, 0x85,
+            0x00, 0x00, 0xa3, 0xe7,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+            0x38, 0x47, 0x75, 0xea,
+            0x00, 0x08, 0x64, 0x70,
+        ];
```

Comment:
If `rustfmt` doesn't complain, maybe what I shared for the other buffer could be used here, as well, if reducing number of lines is a goal. ^^

Reply:
I wouldn't mind, If possible can you share what exactly the goals with regard to formatting and style.
- what is considered (too verbose) ?
- what is considered (too vague) ?

Reply:
I'll leave that to the other reviewers to give the final word. My take is: I go with rustfmt tells me to do. We enforce format style on the c side with clang-format, and if I recall correctly, we want to do the same with rust, with rustfmt, so...

Another general rule that I notice is that it is preferred to have fewer lines of code, when possible.